README.md

```diff
@@ -1,3 +1,195 @@
+
+```python
+from envoy_client.client import AggregatorClient
+from envoy_client.transport import RequestsTransport
+from envoy_client.auth import ClientCerticateAuth
+
+cert_path = '/path/to/client.cert'
+# The aggregator LFDI (long-form device identifier) is derived
+# from the client certificate and will be supplied to the aggregator/client
+aggregator_lfdi = '0x21352135135'  
+
+client = AggregatorClient(
+    transport=RequestsTransport(
+        'https://server-location', 
+        auth=ClientCertificateAuth(
+            cert=cert_path
+        )),
+    lfdi=aggregator_lfdi
+)
+```
+
+## Registration
+
+Under the 2030.5 server model, registration of a device requires a non-neglible number of 
+requests. The registration process requires all or most of these:
+- create `EndDevice`
+- add `DeviceInformation` to the `EndDevice`
+- add associated `DER` containers to the `EndDevice`
+- populate `DERCapability` for each associated `DER`
+- register the `EndDevice` to an associated `ConnectionPoint`
+
+These steps are explained in more detail below.
+
+### POST EndDevice
+
+This registers the `EndDevice` with the server, which corresponds (essentially) to the site
+controller. Note that under the aggregator model, the aggregator is also an `EndDevice`.
+
+> Note: The long-form device identifier (LFDI) is a 160-bit unique identifier that is normally 
+displayed as a hexadecimal string. For `EndDevice`s that use a TLS certificate to communicate
+with the server (e.g. the aggregator itself), this is derived from the certificate fingerprint.
+For `EndDevice`s that do not have a certificate issued, (e.g. aggregator-managed controllers),
+this should correspond to a globally unique identifier that the aggregator already uses, 
+appropriately padded or truncated. (Note that, since the short-form device identifier (SFDI) 
+is based on the left-truncated 36 bits of the LFDI, the LFDI should not be left-padded).
+
+> When using the models in this library, the SFDI is automatically calculated from the supplied
+LFDI. If implemented separately, use the following information:
+
+> The SFDI SHALL be the certificate fingerprint left-truncated to 36 bits. For display purposes, this SHALL be expressed as 11 decimal (base 10) digits, with an additional sum-of-digits checksum digit right- concatenated. For example:
+LFDI left truncated to 36 bits: 0x3E4F45AB3
+Expressed as a decimal: 16726121139
+Checksum added: 167261211391
+
+
+#### Example
+```
+POST https://server-location/edev
+Content-Type: application/xml
+
+<?xml version="1.0" encoding="utf-8"?>
+<EndDevice>
+        <deviceCategory>262144</deviceCategory>
+        <lFDI>0x21352135135</lFDI>
+        <sFDI>356563223726</sFDI>
+        <changedTime>0</changedTime>
+        <postRate>0</postRate>
+        <enabled>true</enabled>
+</EndDevice>
+
+```
+
+The server responds with a `201: CREATED` response and a header `location: /edev/3`
```

Comment:
Note - I need to do a review on all the HTTP code responses to make sure they're correct and consistent.